R/setup.R

```diff
@@ -455,17 +415,16 @@ AgProductionTechnology_setup <- function(aLandLeaf, aAgData, aScenarioInfo) {
       aLandLeaf$mNonLandCostTechChange[[per]] <- 0
     }
 
-    # Filter cost for current year
-    cost %>%
-      filter(year == y) ->
-      currCost
-
     # Set cost in the LandLeaf
-    if(nrow(currCost) & aScenarioInfo$mUseZeroCost == FALSE) {
-      aLandLeaf$mCost[[per]] <- as.numeric(currCost[[c("nonLandVariableCost")]])
+    if(aScenarioInfo$mUseZeroCost == FALSE &
```

Comment:
Single `&` is for combining logical vectors; it does not short-circuit.  Double `&&` is for scalar values; it does short-circuit and is the preferred choice in the conditional of an if statement (which needs a single logical value; it will choke on a logical vector).

```suggestion
    if(aScenarioInfo$mUseZeroCost == FALSE &&
```